wordpress-mobile/WordPress-iOS
# Fixes AccountService unit tests to avoid modifying NSUserDefaults.

### Description:

Running the unit tests would modify the default WP.com account in NSUserDefaults.  That would result in the list of blog being empty when pressing the "Edit" button as shown below.

This PR fixes the unit tests so that they don't affect or persist any changes to the defaults.

### Screenshots:

An example showing the list of blogs broken after running our unit tests:

![emptybloglist](https://user-images.githubusercontent.com/1836005/37116100-ef380bfe-222b-11e8-832a-4b4729c4470b.gif)

An example showing the Me section broken after running our unit tests:

WORK-IN-PROGRESS

### Testing:

**Test 1:**

- Erase your simulator content and settings.
- Run the app.
- Log in manually.
- Go to the list of Blogs and tap "Edit".  Make sure the editable list isn't empty.
- Stop the App and run the unit tests.
- Once done, run the app again and go to the Blogs list.
- Tap "Edit" and make sure the list isn't empty.

**Test 2:**

- Erase your simulator content and settings.
- Run the app.
- Log in manually.
- Stop the App and run the unit tests.
- Once done, run the app again and go to the Me section.
- Make sure your account is still logged in.

Feel free to repeat the process in `develop` to reproduce the original issue.
